### ISSUE TITLE
Forrest plots

### ISSUE THREAD

**Issue by idrdex** [idrdex/star-django#30]
We should implement Forrest plots on the site.  I added R code called by python here: https://github.com/idrdex/star_api/blob/master/Meta_analysis.ipynb

**Comment by Suor** [comment]
Is using R `meta` package is really needed? We have python meta-analysis implementation.

**Comment by idrdex** [comment]
The graphing is in the R meta package.  Otherwise (and ultimately) we have to implement our own python graphing.

**Comment by Suor** [comment]
But can we use our own meta-analysis? while still using R `forest()`.

**Comment by Suor** [comment]
I want to reduce R usage to a minimum.

**Comment by idrdex** [comment]
Nope!  Its built into the R meta package.  So unfortunately we have to run 2 meta-analyses.

**Comment by idrdex** [comment]
Yeah me too.  This will have to be a stop gap measure until we generate the forrest plot natively.

**Comment by Suor** [comment]
I don't understand what does it means it's built into R meta package. Can't we write:
```python
def plot_gene(mygene_sym, fc):#myGeneSym = "ARG1"
    meta_gene = fc[fc.mygene_sym == mygene_sym].drop_duplicates(['gpl','gse'])
    meta_gene.title = mygene_sym
    m = get_gene_meta(meta_gene) # Our meta-analysis
    %R -i m -w 750 -h 500 forest(m, pred=T)
```

**Comment by idrdex** [comment]
No.  It has to be a R native object generated by the R meta package.  Its can’t be python / pandas data frame which is what get_gene_meta returns.

**Comment by idrdex** [comment]
must be this R call: http://www.inside-r.org/packages/cran/meta/docs/metacont
And it generates an R object.

**Comment by Suor** [comment]
Unforturnately I can't find any implementation of forest plots in Python or JavaScript. The closest thing I found is [box plots](http://bl.ocks.org/mbostock/4061502).

**Comment by idrdex** [comment]
Try seaborn for all of those: http://stanford.edu/~mwaskom/software/seaborn/generated/seaborn.boxplot.html#seaborn.boxplot

A box plot will probably work well enough.  Also check out dot_plots in stats_models: http://nbviewer.ipython.org/urls/umich.box.com/shared/static/oh717lkxczhseep71lao.ipynb

dex*

**Comment by idrdex** [comment]
This will bring it to d3js: http://mpld3.github.io

**Comment by Suor** [comment]
Forest plots now generated on click on gene_sym in results table.

**Comment by idrdex** [comment]
We should implement this ASAP in js.

**Comment by idrdex** [comment]
We need to revisit this in d3.js ASAP.

**Comment by dhimmel** [comment]
Check out vega-lite, a JSON specification for web visualization. It will be way easier than writing the raw d3 code directly. You can also use the [altair package](https://github.com/altair-viz/altair) in Python to create vega-lite visualizations.

**Comment by Suor** [comment]
It won't be easier, it doesn't have forest plots implementation and so that needs to be implemented from scratch anyway. It's also more limiting as we won't be able to have any interactivity.

**Comment by Suor** [comment]
These are now deployed.

**Comment by idrdex** [comment]
Looks like the display is a little off on resize.  See right of this screenshot.  The white backgrounds scales smaller than the chart.

![stargeo_meta shot](https://cloud.githubusercontent.com/assets/1265923/20845822/a87e0350-b87b-11e6-8817-3100ed3982b5.png)

**Comment by idrdex** [comment]
Broken for big graphs?  Not working for genes wit big k in http://stargeo.org/analysis/208/

**Comment by Suor** [comment]
It doesn't work for aging because that one doesn't have source data frame
(very old analysis). You may just rerun it.

**Comment by Suor** [comment]
Made plots more compact horizontally and added scrolling when it doesn't fit anyway.

**Comment by Suor** [comment]
And plots don't automatically resize if you resize browser window for now.

**Comment by idrdex** [comment]
Ok. Looks good on phone too. Nice. Can we have URL states for the analysis? How to share a chart or configuration of tabs?

**Comment by ir4y** [comment]
@idrdex I this we can store this information at URL anchor.
It will be something like this
`http://stargeo.org/analysis/334/#0610005C13Rik,0610009B22Rik`
Tomorrow I will start working on this feature.

**Comment by Suor** [comment]
So sharing is on stargeo, thx to @ir4y. To share a chart open it and share URL, same with several charts.

**Comment by idrdex** [comment]
Thx @ir4y and @Suor.  Looks awesome! Almost working flawlessly.  Data table needs to be refreshed when accessing for the first time with url.  For instance see: http://stargeo.org/analysis/336/#LTF,ELANE
Graphs are ok, but results data table is empty until I toggle a sort order or a header. (In safari on a mac at least)

**Comment by idrdex** [comment]
Found plots out of sync on rerun: http://stargeo.org/analysis/339/#Tgtp1

k=3 but only 2 studies graphed. I reran the analysis after tagging a new study. It's not really an issue because we should just retire rerun in #58

**Comment by Suor** [comment]
@idrdex results table loads for me. Is it a consistent issue for you? Maybe it was just a network glitch?

**Comment by Suor** [comment]
Graph data is cached for several days unconditionally, so yes rerunning analysis leads to that bug. Just removed inplace rerun, so it should not be an issue anymore.

**Comment by ir4y** [comment]
@idrdex 
Now the data table refreshing when accessing for the first time with url. It was a MacOS specific bug.
It is fixed and deployed.

**Comment by idrdex** [comment]
Graph labels get muddles on small views.  On my iPhone for sure.  And on Safari when window size is limited.  Can we fix this?

**Comment by horberlan** [comment]
See https://github.com/horberlan/forest-plot
